goaviatrix/security_domain.go

```diff
@@ -207,6 +222,31 @@ func (c *Client) DisableIntraDomainInspection(ctx context.Context, intraDomainIn
 	return c.PostAPIContext(ctx, params["action"], params, BasicCheck)
 }
 
+func (c *Client) GetAllSecurityDomains() ([]NetworkDomainDetails, error) {
+	params := map[string]string{
+		"action": "list_all_tgw_security_domains",
+		"CID":    c.CID,
+	}
+
+	type DomainDetail struct {
+		Domains []NetworkDomainDetails `json:"domains"`
+	}
+
+	type Resp struct {
+		Return  bool         `json:"return"`
+		Results DomainDetail `json:"results"`
+		Reason  string       `json:"reason"`
+	}
+
+	var data Resp
+	err := c.GetAPI(&data, params["action"], params, BasicCheck)
+	if err != nil {
+		return nil, err
+	}
+	domainList := data.Results.Domains
```

Comment:
Small nit: this assignment does nothing. You can just `return data.Results.Domains, nil`

Reply:
have fixed